src/tribol/mesh/MeshData.cpp

```diff
@@ -226,207 +194,277 @@ bool MeshElemData::isValidRatePenalty( PenaltyEnforcementOptions& pen_options )
 // Routines for MeshData class //
 //                             //
 /////////////////////////////////
-MeshData::MeshData()
-  : m_elementType( tribol::UNDEFINED_ELEMENT )
-  , m_lengthNodalData(0)
-  , m_numSurfaceNodes(0)
-  , m_numCells(0)
-  , m_numNodesPerCell(0)
-  , m_isValid(true)
-  , m_positionX(nullptr), m_positionY(nullptr), m_positionZ(nullptr)
-  , m_dispX(nullptr), m_dispY(nullptr), m_dispZ(nullptr)
-  , m_forceX(nullptr), m_forceY(nullptr), m_forceZ(nullptr)
-  , m_velX(nullptr), m_velY(nullptr), m_velZ(nullptr)
-  , m_nX(nullptr), m_nY(nullptr), m_nZ(nullptr)
-  , m_cX(nullptr), m_cY(nullptr), m_cZ(nullptr)
-  , m_faceRadius(nullptr)
-  , m_area(nullptr)
-  , m_connectivity(nullptr)
-  , m_sortedSurfaceNodeIds(nullptr)
-  , m_node_nX(nullptr), m_node_nY(nullptr), m_node_nZ(nullptr)
+MeshData::MeshData( IndexT mesh_id, IndexT num_elements, IndexT num_nodes,
+                    const IndexT* connectivity, InterfaceElementType element_type,
+                    const RealT* x, const RealT* y, const RealT* z,
+                    MemorySpace mem_space )
+  : m_mesh_id( mesh_id )
+  , m_element_type( element_type )
+  , m_dim( getDimFromElementType() )
+  , m_num_nodes( num_nodes )
+  , m_mem_space( mem_space )
+  , m_allocator_id( getResourceAllocatorID(mem_space) )
+  , m_is_valid( true )
+  , m_position( createNodalVector(x, y, z) )
+  , m_connectivity( createConnectivity(num_elements, connectivity) )
 {
+  // mesh verification
```

Comment:
The initialization was previously done in `tribol::registerMesh()`.  The checks are changed slightly, but not in a meaningful way.  I tried to simplify it a bit.  First, the dimension is set from the element type instead of if z is a nullptr.  Also, there was an element type check.  I removed this, since this probably makes more sense to check at the method/coupling scheme level.